python/cudf/cudf/core/dataframe.py

```diff
@@ -2123,6 +2134,10 @@ def loc(self):
     def iloc(self):
         """
         Selecting rows and column by position.
+        In case result is expected to be a single
```

Comment:
This isn't always true though. It's possible to get a single row as a `DataFrame`. See Pandas behaviour (which I think we emulate, but can't confirm right now):

```python

In [4]: a.iloc[0, :]  # single row returned as a Series                 
Out[4]: 
a    1
b    2
Name: 0, dtype: int64

In [5]: a.iloc[[0,], :] # single row returned as a DF                     
Out[5]: 
   a  b
0  1  2
```

I would love for us to document the behaviour of `loc`/`iloc` w.r.t downcasting, but be warned that it's full of quirky edge cases and we don't always match Pandas behaviour. :-)

Reply:
Is it for internal purpose or do you think it would be better with having it in documentation?

If it is for internal purpose, we can add test cases for all supporting edge cases.

Reply:
And I can see most of them have been taken care in test cases.